### DISCUSSION TITLE
Resolving array of IDs into objects, based on indentifiers and references

### DISCUSSION THREAD

**Discussion by esbenvb** [mobxjs/mobx-state-tree#1701] [Q&A] [answered]
I have a store where I load in a list of authors, which is the type `EgInteressent`

The `BulletinBoard` contains a list of `authors` which are of the type `EgInteressent`.

The `BulletinBoardPost` contains an `author` field with the ID of an author and successfully resolves that ID into an element from the `authors` list of the `BulletinBoard`.

Now I want to add another field, `likedBy` which contains an array of IDs that should all be resolved into `EgInteressent` objects.

But the code I've tried to implement, does not resolve the IDs of the `likedBy`, as you can see in the comments.

What am I doing wrong?

```ts
// Props that both `BulletinBoardPost` and `BulletinBoardPostComment` share
export const BulletinBoardPostSharedProps = {
  postId: types.identifier,
  author: types.safeReference(EgInteressent), // this resolves to {"egInteressentNummer": "434343", "name": "Bobby Olsen"}
  body: types.string,
  likedBy: types.array(types.safeReference(EgInteressent)), // This remains like  ["454545", "434343"]
  files: types.maybe(types.array(File)),
  createDate: types.string,
}
```

```ts
export const EgInteressent = types.model('EgInteressent').props({
  egInteressentNummer: types.identifier,
  name: types.string,
})
```

```ts
export const BulletinBoard = types.model('BulletinBoard').props({
  boardId: types.identifier,
  name: types.string,
  authors: types.optional(types.map(EgInteressent), {}),
  posts: types.optional(types.array(BulletinBoardPost), []),
  isExhausted: types.optional(types.boolean, false),
  isLoading: types.optional(types.boolean, false),
})
```

**Comment by esbenvb** [comment, marked as answer]
I made it work. So apparently, I cannot console.log the values of a property of the store, I have to iterate through the array before I get the resolved values.

https://codesandbox.io/s/mobx-state-tree-todolist-forked-u53bu

**Reply by BATCOH** [reply]
You made an example with really old MobX/MST versions.
I made a fork with updated versions: https://codesandbox.io/s/mobx-state-tree-todolist-forked-5ge7i?file=/index.js

And everything looks fine, including console.log.
